### ISSUE TITLE
torchscript version runs out of CUDA memory in c++ but it's ok in pytorch

### ISSUE THREAD

**Issue by zhouhuirun2015** [ZHKKKe/MODNet#100]
Any idea？Thanks.

**Comment by zhouhuirun2015** [comment]
The following operation failed in the TorchScript interpreter. Traceback of TorchScript, serialized code (most recent call last): File "code/__torch__/src/models/backbones/mobilenetv2/___torch_mangle_10.py", line 14, in forward _0 = _1 else: _0 = (self.conv).forward(x, ) ~~~~~~~~~~~~~~~~~~ <--- HERE return _0 File "code/__torch__/torch/nn/modules/container/___torch_mangle_9.py", line 25, in forward input0 = (_0).forward(input, ) input1 = (_1).forward(input0, ) input2 = (_2).forward(input1, ) ~~~~~~~~~~~ <--- HERE input3 = (_3).forward(input2, ) input4 = (_4).forward(input3, ) File "code/__torch__/torch/nn/modules/container/___torch_mangle_9.py", line 24, in forward _7 = getattr(self, "7") input0 = (_0).forward(input, ) input1 = (_1).forward(input0, ) ~~~~~~~~~~~ <--- HERE input2 = (_2).forward(input1, ) input3 = (_3).forward(input2, ) File "code/__torch__/torch/nn/modules/batchnorm/___torch_mangle_5.py", line 28, in forward else: bn_training = False _3 = _0(input, self.running_mean, self.running_var, self.weight, self.bias, bn_training, 0.10000000000000001, 1.0000000000000001e-05, ) ~~ <--- HERE return _3 def _check_input_dim(self: __torch__.torch.nn.modules.batchnorm.___torch_mangle_5.BatchNorm2d, File "code/__torch__/torch/nn/functional.py", line 14, in batch_norm else: pass _2 = torch.batch_norm(input, weight, bias, running_mean, running_var, training, momentum, eps, True) ~~~~~~~~~~~~~~~~ <--- HERE return _2 def hardtanh(input: Tensor, Traceback of TorchScript, original code (most recent call last): File "C:\Users\admin\AppData\Roaming\Python\Python38\site-packages\torch\nn\modules\container.py", line 117, in forward def forward(self, input): for module in self: input = module(input) ~~~~~~ <--- HERE return input File "C:\Users\admin\AppData\Roaming\Python\Python38\site-packages\torch\nn\modules\container.py", line 117, in forward def forward(self, input): for module in self: input = module(input) ~~~~~~ <--- HERE return input File "C:\Users\admin\AppData\Roaming\Python\Python38\site-packages\torch\nn\modules

**Comment by zhouhuirun2015** [comment]
put torch::NoGradGuard no_grad; before forward.
:)
